DDDEastAnglia.Tests/Services/Messengers/Email/EmailMessengerFacts.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Net.Mail;
+using DDDEastAnglia.Helpers;
+using DDDEastAnglia.Models;
+using DDDEastAnglia.Services.Messenger.Email;
+using NSubstitute;
+using NUnit.Framework;
+using MailMessage = DDDEastAnglia.Services.Messenger.Email.MailMessage;
+
+namespace DDDEastAnglia.Tests.Services.Messengers.Email
+{
+    [TestFixture]
+    public class EmailMessengerFacts
+    {
+        [TestFixture]
+        public class Notify_Should
+        {
+            [Test]
+            public void Prepare_The_Mail_For_Delivery()
+            {
+                var postman = Substitute.For<IPostman>();
+                var template = Substitute.For<IMailTemplate>();
+                template.RenderBody().Returns("Message body");
+                template.RenderSubjectLine().Returns("Message subject");
+
+                var messenger = new EmailMessenger(postman, template);
+                messenger.Notify(new UserProfile {EmailAddress = "speaker@dddeastanglia.com"});
+
+                var message = new MailMessage
+                {
+                    From = new MailAddress("admin@dddeastanglia.com", "DDD East Anglia"),
```

Comment:
It *is* this test :smile: This is a collaboration test*, ensuring the `EmailMessenger` calls the `IPostman` correctly: is the correct method on `IPostman` called, and is it passed the correct arguments?

The `EmailMessenger` is responsible for preparing a `MailMessage` to be passed to its `IPostman` collaborator. There will only be this one implementation of `EmailMessenger`, even if we were to switch to something other than SendGrid as that responsibility is abstracted by the `IPostman`. The test asserts that `IPostman.Deliver()` is called with the expected `MailMessage` instance; that instance includes the `From` address details. Therefore, the test does correctly assert that all emails are sent from "DDD East Anglia \<admin@dddeastanglia.com\>".

\* See [Integration Tests are a Scam](https://vimeo.com/80533536) by J.B. Rainsberger, *well* worth a watch.